R/helper_clean_model_data.R

```diff
@@ -49,6 +52,19 @@ clean_model_data <- function(data, datargs, estimator = "") {
     mfargs[["fixed_effects"]] <- sym(name)
   }
 
+  recycle_prs <- FALSE
+  if ("condition_pr" %in% names(mfargs)) {
+    condition_pr <- eval_tidy(mfargs[["condition_pr"]], data = data)
+    if (length(condition_pr) == 1) {
+      recycle_prs <- TRUE
+      mfargs[["condition_pr"]] <- NULL
+    } else {
+      name <- sprintf(".__condition_pr%%%d__", sample.int(.Machine$integer.max, 1))
+      m_formula_env[[name]] <- condition_pr
+      mfargs[["condition_pr"]] <- sym(name)
```

Comment:
Let the loop above do this part (else block), this block can clean it up if it's length == 1.

Reply:
will fix